CC3D/Models/BiocIU/SARSCoV2MultiscaleVTM/Model/Models/DrugDosingModel/DrugDosingModelSteppable.py

```diff
@@ -0,0 +1,215 @@
+                   _dose_interval, _dose)
+
+        drug_dosig_model_vars = ["Drug", "Available1", "Available2", "Available3", "Available4"]
+
+        return dosingmodel_str, drug_dosig_model_vars
+
+    def init_plots(self):
+        self.ddm_data_win = self.add_new_plot_window(title='Drug dosing model',
+                                                     x_axis_title='Time (hours)',
+                                                     y_axis_title='Variables',
+                                                     x_scale_type='linear',
+                                                     y_scale_type='linear',
+                                                     grid=True,
+                                                     config_options={'legend': True})
+        colors = ['blue', 'red', 'green', 'yellow', 'white']
+        for c, var in zip(colors, self.ddm_vars):
+            self.ddm_data_win.add_plot(var, style='Dots', color=c, size=5)
+
+        self.rmax_data_win = self.add_new_plot_window(title='r_max vs Time',
+                                                      x_axis_title='Time (hours)',
+                                                      y_axis_title='r_max',
+                                                      x_scale_type='linear',
+                                                      y_scale_type='linear',
+                                                      grid=True,
+                                                      config_options={'legend': True})
+
+        self.rmax_data_win.add_plot('rmax', style='Dots', color='red', size=5)
+
+    def start(self):
+
+        # set model string
+        self.drug_model_string, self.ddm_vars = self.set_drug_model_string(Drug, Available1, Available2, Available3,
+                                                                           Available4,
+                                                                           k0, d0, k1, d1, k2, d2, k3, d3, d4,
+                                                                           first_dose,
+                                                                           initial_dose, dose_interval, dose)
+
+        # init sbml
+        self.add_free_floating_antimony(model_string=self.drug_model_string, step_size=days_2_mcs,
+                                        model_name='drug_dosing_model')
+
+        # init plots
+        if self.plot_ddm_data:
+            self.init_plots()
+
+        # init save data
+        if self.write_ddm_data:
+            from pathlib import Path
+            for key, rel_path in self.data_files.items():
+                self.data_files[key] = Path(self.output_dir).joinpath(rel_path)
+                with open(self.data_files[key], 'w'):
+                    pass
+
+        # Post reference to self
+        self.shared_steppable_vars[self.drug_dosing_model_key] = self
+
+    def get_rmax(self, avail4):
```

Comment:
Recommend making the target viral replication model variable an input to this module. This would facilitate usage of this module for investigating other target mechanisms besides the one currently implemented.